File: node/api-tutorial.js
const request = require('request');

const apiKey = "SET-YOUR-API-KEY-HERE";

function getFullURL(url, options){
    const params = []
    for (let key in options) {
        params.push(`${key}=${options[key]}`)
    }
    return url + '?' + params.join("&");
}

const baseURL = "https://min-api.cryptocompare.com/data/price";

const options = {
    api_key: apiKey,
    fsym: "BTC",
    tsyms: "USD"
};

const fullURL = getFullURL(baseURL, options);

request.get(fullURL, function(err, res, body){
    if (err){
        console.log(err);
    } else {
        console.log(body);
    }
});

File: python/api-tutorial.py
import requests

def getUrlToJSON(url):
    return requests.get(url).json()
    
def generateURL(endpoint, options):
    params = []
    for key in options:
        params.append("{}={}".format(key, options[key]))
    return endpoint + "?" + "&".join(params)
    
apiKey = "SET-YOUR-API-KEY-HERE"

endpoint = "https://min-api.cryptocompare.com/data/price"

options = {
    "apiKey": apiKey,
    "fsym": "BTC",
    "tsyms": "USD"
}

url = generateURL(endpoint, options)
result = getUrlToJSON(url)
print(result)

